clang/lib/CodeGen/CGSYCLRuntime.cpp

```diff
@@ -65,8 +65,12 @@ bool CGSYCLRuntime::actOnFunctionStart(const FunctionDecl &FD,
     F.setMetadata("sycl_explicit_simd", llvm::MDNode::get(F.getContext(), {}));
 
   // Set the function attribute expected by the vector backend compiler.
-  if (const SYCLIntelESimdWidenAttr *A = FD.getAttr<SYCLIntelESimdWidenAttr>())
-    F.addFnAttr("CMGenxSIMT", std::to_string(A->getNumber()));
+  if (const auto *A = FD.getAttr<SYCLIntelESimdVectorizeAttr>())
+    if (const auto *DeclExpr = dyn_cast<ConstantExpr>(A->getValue())) {
```

Comment:
Since the argument isn't optional I'm not sure if we need this check. I think this can be replaced by an assert @AaronBallman

Reply:
Yup -- the `dyn_cast` should be a `cast` because by CodeGen time we've already resolved any dependent expressions.

Reply:
Done